pyomo/contrib/mindtpy/util.py

```diff
@@ -181,3 +181,60 @@ def var_bound_add(solve_data, config):
                         var.setub(config.integer_var_bound)
                     else:
                         var.setub(config.continuous_var_bound)
+
+
+def generate_L2_objective_function(model, setpoint_model, discretes_only=False):
```

Comment:
For this whole file, I think we can be a bit more explicit with the function names. Let's use Norm1 and Norm2 instead of L (it is not clear what we mean with L).